hera_xblocks/question/question/question.py

```diff
@@ -87,17 +114,51 @@ def student_view(self, context=None):
         )
         frag = Fragment(html.format(self=self))
         frag.add_css(self.resource_string("static/css/question.css"))
-        frag.add_javascript_url("https://cdn.jsdelivr.net/gh/vast-engineering/jquery-popup-overlay@2/jquery.popupoverlay.min.js")
         frag.add_javascript(self.resource_string("static/js/src/question.js"))
-        frag.initialize_js('QuestionXBlock')
+
+        frag.add_css_url("https://cdnjs.cloudflare.com/ajax/libs/slick-carousel/1.8.1/slick.css")
+        frag.add_javascript_url("https://cdnjs.cloudflare.com/ajax/libs/slick-carousel/1.8.1/slick.js")
+
+        frag.initialize_js('QuestionXBlock', json_args=self.get_context())
         return frag
-    
+
     @XBlock.json_handler
     def get_data(self, somedata, sufix=''):
         return self.data
 
     @XBlock.json_handler
-    def submit(self, answer, suffix=''):
-        correct_answers = [ option["title"] for option in self.options if option["correct"]==True ]
-        print(answer, correct_answers)
-        return True if answer in correct_answers else False
+    def submit(self, data, suffix=''):
+        user_answers = []
+        correct = False
+        answers = data.get("answers")
+        try:
+            self.user_confidence = int(data.get("confidence"))
+        except ValueError:
+            self.user_confidence = None
+
+        if self.question.get('type') == "number":
+                try:
+                    user_answer = float(answers.get('value'))
+                    user_answers.append(user_answer)
+                    correct_answer = self.answer
+                    if correct_answer - self.preciseness <= user_answer <= correct_answer + self.preciseness:
+                        correct = True
+                except ValueError:
+                    user_answer = None
+
+        elif self.question.get('type') == "text":
+                user_answers.append(answers.get('value'))
+                if answer.get('value') == self.answer:
+                    correct = True
+
+        elif self.question.get('type') in ["select", "radio", "checkbox"]:
+            correct_answers = [ option["title"] for option in self.options if option["correct"] is True ]
+            for answer in answers:
+                user_answers.append(answer.get('value'))
+            if set(user_answers) == set(correct_answers):
```

Comment:
```suggestion
            if set(answers) == set(correct_answers):
```
and we don't need the loop above?